wire/internal/wire/parse.go

```diff
@@ -38,6 +38,37 @@ type providerSetSrc struct {
 	Import   *ProviderSet
 }
 
+// trace returns a slice of strings describing the (possibly recursive) source
+// of p, including line numbers.
+func (p *providerSetSrc) trace(fset *token.FileSet, typ types.Type) []string {
```

Comment:
I also considered adding a String() func to each of the 4 types used here (Provider, ProviderSet, IfaceBinding, Value) and using %v here, thoughts?

Reply:
I'd avoid providing a canonical string function for the exported API for now. This function is fine as it stands.